### ISSUE TITLE
Can't update the name of an OrganizationRole

### ISSUE THREAD

**Issue by James-Ronayne** [p2-inc/keycloak-orgs#58]
At the moment the PUT endpoint for role organisations only allows you to update the the roles description. Is there a reason that updating the role name is not supported?

**Comment by xgp** [comment]
@James-Ronayne thanks for raising this. Because of the way that we have implemented roles, there are relationships elsewhere that are based on the `name`. Because of that, it needs to be immutable once created. For the case where you want to change the role name, we recommend creating a new role with that name, and then adding members to that new role before removing them from the old role (which you can then delete).

**Comment by James-Ronayne** [comment]
Thanks for the speedy reply @xgp! That makes sense if there are other relationships elsewhere. I just saw it had a unique id column so assumed it would try to use the id for those relationships elsewhere, makes sense if not. It's a shame as I have a use case where the role name can change but the user assignments wouldn't. 

If that's the only way to do the rename at the moment it's not the end of the world just a bit suboptimal. While I've got your attention the change your discussing here would be a great addition to see as well https://github.com/p2-inc/keycloak-orgs/issues/48 😄
